src/handlers/baseFieldsHandlers.ts

```diff
@@ -78,7 +78,49 @@ const postBaseField = (
     });
 };
 
+const putBaseField = (
+  req: Request<{ id: string }>,
+  res: Response,
+  next: NextFunction,
+) => {
+  const id = Number.parseInt(req.params.id, 10);
+  if (Number.isNaN(id)) {
+    next(new InputValidationError(
+      'The entity id must be a number.',
+      [],
+    ));
+    return;
+  }
+  const body = req.body as unknown;
+  if (!isBaseFieldWrite(body)) {
+    next(new InputValidationError(
+      'Invalid request body.',
+      isBaseFieldWrite.errors ?? [],
+    ));
+    return;
+  }
+  (async () => {
```

Comment:
The reason for floating this pattern is that it allows us to invoke multiple promises in sequence without having to worry about nested `thens` (basically, let lets us use async / await syntax).  I was intending for this to become the New Way of Doing it.

The root problem here is that we can't have our handlers be async functions since express doesn't expect the handler to be a promise / doesn't handle failed promises.  Using the IFFE lets us treat the bulk of our handler into an async function.

I personally somewhat prefer the idea of having the pattern implemented even in cases where there is only one promise simply because it means if / when we need to add more promises to a given handler we wouldn't need to totally refactor the function / change *all* the lines.

---

That was the reasoning; what do you think!

Reply:
I like it -- unless you disagree I think what makes most sense is to take on either beta 5 OR the wrap function pattern as a separate issue / PR, and just not use an IIFE for this instance since there's only one async call in this case.

Reply:
Created https://github.com/PhilanthropyDataCommons/service/issues/700